lib/smart_answer_flows/calculate-statutory-sick-pay.rb

```diff
@@ -68,17 +72,60 @@ def define
           response
         end
 
+        calculate :sick_end_date_for_awe do |response|
+          response
+        end
+
         next_node_calculation(:days_sick) do |response|
           start_date = sick_start_date
           last_day_sick = response
           (last_day_sick - start_date).to_i + 1
         end
+
         validate { days_sick >= 1 }
-        next_node_if(:paid_at_least_8_weeks?) { days_sick > 3 }
+        next_node_if(:has_linked_sickness?) { days_sick > 3 }
         next_node(:must_be_sick_for_4_days)
       end
 
-      # Question 5.1
+      # Question 6
+      multiple_choice :has_linked_sickness? do
+        option yes: :linked_sickness_start_date?
+        option no: :paid_at_least_8_weeks?
+      end
+
+      # Question 6.1
+      date_question :linked_sickness_start_date? do
+        from { Date.new(2010, 1, 1) }
+        to { Date.today.end_of_year }
+        validate_in_range
+
+        calculate :sick_start_date_for_awe do |response|
+          response
+        end
+
+        next_node(:linked_sickness_end_date?)
+      end
+
+      # Question 6.2
+      date_question :linked_sickness_end_date? do
+        from { Date.new(2010, 1, 1) }
+        to { Date.today.end_of_year }
+        validate_in_range
+
+        calculate :sick_end_date_for_awe do |response|
+          response
+        end
+
+        calculate :prior_sick_days do |response|
```

Comment:
We're not validating if `end date > start date` are we? Perhaps we should? This was not relevant before, because we used to ask for the duration of sickness rather than the end date.